test/schema.uuid.test.js

```diff
@@ -129,6 +129,26 @@ describe('SchemaUUID', function() {
     assert.equal(organization, undefined);
   });
 
+  it('works with populate (gh-13267)', async function() {
+    const userSchema = new mongoose.Schema({
+      _id: { type: 'UUID', default: () => randomUUID() },
+      name: String,
+      createdBy: {
+        type: 'UUID',
+        ref: 'User'
+      }
+    });
+    const User = db.model('User', userSchema);
+
+    const u1 = await User.create({ name: 'admin' });
+    const { _id } = await User.create({ name: 'created', createdBy: u1._id });
+
+    const pop = await User.findById(_id).populate('createdBy');
```

Comment:
maybe consider using `.orFail` here

```suggestion
    const pop = await User.findById(_id).populate('createdBy').orFail();
```